src/tests/units/operator_tests.cpp

```diff
@@ -399,8 +410,8 @@ void tests(DeviceType device, Type floatType = Type::float32) {
       std::vector<float> SV;    // create CSR version of S
       std::vector<IndexType> SI, SO;
       SO.push_back((IndexType)SI.size());
-      for (IndexType i = 0; i < S->shape()[0]; i++) {
-        for (IndexType j = 0; j < S->shape()[1]; j++) {
+      for (IndexType i = 0; size_t(i) < S->shape()[0]; i++) {
```

Comment:
Shape has `int` value type currently, so would be better to have the comparison look like this: `i < (IndexType)S->shape()[0]`

Reply:
Fixed.